Content.Server/Store/Systems/StoreSystem.cs

```diff
@@ -67,6 +69,12 @@ private void OnAfterInteract(EntityUid uid, CurrencyComponent component, AfterIn
         if (args.Target == null || !TryComp<StoreComponent>(args.Target, out var store))
             return;
 
+        // require the store to be open before inserting currency
+        var user = args.User;
+        Logger.DebugS("TEMP_error", $"open: {_ui.SessionHasOpenUi(uid, StoreUiKey.Key, Comp<ActorComponent>(user).PlayerSession)}");
+        if (!TryComp<ActorComponent>(user, out var actor) || !_ui.SessionHasOpenUi(uid, StoreUiKey.Key, actor.PlayerSession))
+            return;
```

Comment:
Now that uplinks depend on ringtone you can just check for that instead right? (also check if not-pda uplinks still work)

Reply:
i think that would instead just be integrated with uplink locking/unlocked if/when its done, makes mores sense than explicitly checking ringtone here